packages/layout/src/node/shouldBreak.js

```diff
@@ -55,7 +64,65 @@ const getWrapTextAroundChildrenHeight = node => {
   return height;
 };
 
-const shouldBreak = (child, futureElements, height) => {
+/**
+ * Returns whether the node has any drawable lines
+ *
+ * @param {Array} content
+ * @returns boolean
+ */
+const hasAnyLines = content => {
+  if (Array.isArray(content)) {
+    for (const child of content) {
+      if (child?.lines?.length) {
+        return true;
+      }
+
+      return hasAnyLines(child?.children);
+    }
+  }
+
+  return false;
+};
+
+/**
+ * Returns whether the node should be pushed to the next page
+ * by checking the behaviour of next sibling
+ *
+ * @param {Array} futureElements
+ * @param {Number} presenceAhead
+ * @param {Number} height
+ * @param {Number} contentArea
+ * @returns boolean
+ */
+const shouldPushToNextPage = (
```

Comment:
Can we change the name of this function to reflect what it does?
When considered in isolation, the name does not give information about what's happening here